### DISCUSSION TITLE
Allow collection of buyer name or other information

### DISCUSSION THREAD

**Discussion by jonathanalevi** [btcpayserver/btcpayserver#2460] [General]
Hi all
I’ve successfully implemented the modal on my site, http://becomeasuperlearner.com, which I then webhook to Zapier to process the order... however, I really need to be able to collect the users name to make an account for them... I saw this https://github.com/btcpayserver/btcpayserver/issues/939 but it seems no progress has been made since 2019. How can I do this without knowing how to code? It seems like an obvious must have to me...

**Comment by vsatmydynipnet** [comment]
+1 I am in need for collecting the buyer information using the pay button. Otherwise I am forced to add PHP, scripts and other insecure stuff to a pure static html server, serving websites. to be able to add the full buyer information fields in the button opening btcpay server would be a massive game changer for people running html/static servers/sites only.

**Comment by ghost** [comment]
This is useful as a opt-in feature, probably a perfect fit for a plug-in.

**Comment by bolatovumar** [comment]
@entreprenewer couldn't you just use their email as a username? You can collect emails from invoices.

![Screen Shot 2021-04-17 at 5 57 41 PM](https://user-images.githubusercontent.com/1934678/115130693-7a4a2880-9fa6-11eb-9977-0bd3433f60fa.png)

**Reply by ghost** [reply]
Well, it would be useful to have multiple different potential options to pick from here:
- email
- name
- shipping address
- twitter
- telegram
- matrix
- nextcloud
etc.

**Reply by Zaxounette** [reply]
The POS and other already available `apps` will probably migrate to plugins when they're available.
This will be possible then.

A lot of things will be possible with plugins. A whole lot !

**Reply by ghost** [reply]
but wen plugins sir? :laughing:

**Reply by Zaxounette** [reply]
soon™ :laughing:

**Reply by Zaxounette** [reply]
More seriously, here's the 2021 roadmap: https://github.com/orgs/btcpayserver/projects/9

It's an approximation, if some features don't make it in one quarter, they'll be moved to the next.

**Reply by jonathanalevi** [reply]
> @entreprenewer couldn't you just use their email as a username? You can collect emails from invoices.
> 
> ![Screen Shot 2021-04-17 at 5 57 41 PM](https://user-images.githubusercontent.com/1934678/115130693-7a4a2880-9fa6-11eb-9977-0bd3433f60fa.png)

For email marketing, and creating an account on the course player. I need their first and last name. 
Imagine spending ~$1,000 on a product, only to receive emails addressed to "Hey there, Unknown" or seeing your profile set up in the top right as "Unknown P." - not a great user experience ;)

**Reply by jonathanalevi** [reply]
> More seriously, here's the 2021 roadmap: https://github.com/orgs/btcpayserver/projects/9
> 
> It's an approximation, if some features don't make it in one quarter, they'll be moved to the next.

I don't suppose donations help with this timeline, do they? ;)

**Comment by vsatmydynipnet** [comment]
@bolatovumar e-mail is no option here. i already found a workaround building a script sending data via Element Matrix Messenger. So temporary it is ok. here. But collenctiong buyer info should work  and would make things easier.

**Comment by pavlenex** [comment]
https://docs.btcpayserver.org/Forms/
